bigtable/bttest/example_test.go

```diff
@@ -16,27 +16,59 @@ limitations under the License.
 package bttest_test
 
 import (
+	"cloud.google.com/go/bigtable"
 	"context"
 	"fmt"
-	"log"
-
-	"cloud.google.com/go/bigtable"
 	"github.com/fullstorydev/emulators/bigtable/bttest"
+	"github.com/testcontainers/testcontainers-go"
 	"google.golang.org/api/option"
 	"google.golang.org/grpc"
+	"log"
 )
 
-func ExampleNewServer() {
-
+func ExampleLocalServer() {
 	srv, err := bttest.NewServer("localhost:0")
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	validateServer(srv.Addr)
+}
+
+func ExampleContainerServer() {
+	ctx := context.Background()
+	req := testcontainers.ContainerRequest{
+		Image:        "cbtemulator",
+		ExposedPorts: []string{"9000"},
+	}
+	c, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	if err != nil {
+		log.Fatalln(err)
+	}
+	defer c.Terminate(ctx)
 
+	ip, err := c.Host(ctx)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	port, err := c.MappedPort(ctx, "9000")
 	if err != nil {
 		log.Fatalln(err)
 	}
 
+	srvAddr := fmt.Sprintf("%s:%s", ip, port.Port())
```

Comment:
This is how we get the port when using testcontainers

Reply:
nice